renjin/org/renjin/cran/stringi/stringi.java

```diff
@@ -82,7 +91,25 @@ public static void R_init_stringi(DllInfo dll) {
   public static SEXP stri_enc_set(SEXP s1) { throw new EvalException("TODO"); }
   public static SEXP stri_enc_fromutf32(SEXP s1) { throw new EvalException("TODO"); }
   public static SEXP stri_enc_toascii(SEXP s1) { throw new EvalException("TODO"); }
-  public static SEXP stri_enc_toutf8(SEXP s1, SEXP s2, SEXP s3) { throw new EvalException("TODO"); }
+  public static SEXP stri_enc_toutf8(SEXP str, SEXP is_unknown_8bit, SEXP validate) {
+    final boolean strict_ascii = is_unknown_8bit.getElementAsSEXP(0).asLogical().toBooleanStrict();
+    final int length = str.length();
+    final String[] result = new String[length];
+
+    for (int i = 0; i < length; i++) {
+      final String element = str.getElementAsSEXP(i).asString();
```

Comment:
I would normally cast `str` to an `AtomicVector`, and then you can call `getElementAsString(i)` and avoid creating an new SEXP for each element.